modules/platforms/dotnet/Apache.Ignite.Core/Binary/BinaryBasicNameMapper.cs

```diff
@@ -125,5 +157,27 @@ public string GetFieldName(string name)
 
             return sb;
         }
+
+        /// <summary>
+        /// Perform mapping to java naming convention e.g `Com.Company.Class` maps to `com.company.Class`.
+        /// </summary>
+        /// <param name="name">Type name with the namespace.</param>
+        /// <param name="javaDomain">Java domain to be added.</param>
+        /// <returns></returns>
+        private static string DoForceJavaNamingConventions(string name, string javaDomain)
+        {
+            if (javaDomain != null)
+            {
+                name = javaDomain + name;
+            }
+
+            var arr = name.ToCharArray();
```

Comment:
I think we can simplify lowering logic if we add `TypeNameParser.GetNamespace` method, something like:
`p.GetNamespace.ToLower(CultureInfo.InvariantCulture) + '.' + p.GetName()`.

Reply:
Done.